Dockerfile

```diff
@@ -0,0 +1,333 @@
+           yarn \
+    && apt-get autoremove -yqq --purge \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install MySQL client from Oracle repositories (Debian installs mariadb)
+RUN KEY="A4A9406876FCBD3C456770C88C718D3B5072E1F5" \
+    && GNUPGHOME="$(mktemp -d)" \
+    && export GNUPGHOME \
+    && for KEYSERVER in $(shuf -e \
+            ha.pool.sks-keyservers.net \
+            hkp://p80.pool.sks-keyservers.net:80 \
+            keyserver.ubuntu.com \
+            hkp://keyserver.ubuntu.com:80 \
+            pgp.mit.edu) ; do \
+          gpg --keyserver "${KEYSERVER}" --recv-keys "${KEY}" && break || true ; \
+       done \
+    && gpg --export "${KEY}" | apt-key add - \
+    && gpgconf --kill all \
+    rm -rf "${GNUPGHOME}"; \
+    apt-key list > /dev/null \
+    && echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-5.7" | tee -a /etc/apt/sources.list.d/mysql.list \
+    && apt-get update \
+    && apt-get install --no-install-recommends -y \
+        libmysqlclient-dev \
+        mysql-client \
+    && apt-get autoremove -yqq --purge \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# disable bytecode generation (only in build image - it does not affect the main image)
+ENV PYTHONDONTWRITEBYTECODE=1
+
+ARG PIP_VERSION
+ENV PIP_VERSION=${PIP_VERSION}
+
+RUN pip install --upgrade pip==${PIP_VERSION}
+
+ARG AIRFLOW_SOURCES_TO="/opt/airflow"
+ENV AIRFLOW_SOURCES_TO=${AIRFLOW_SOURCES_TO}
+
+COPY ${AIRFLOW_SOURCES_FROM} ${AIRFLOW_SOURCES_TO}
+
+ARG CASS_DRIVER_BUILD_CONCURRENCY
+ENV CASS_DRIVER_BUILD_CONCURRENCY=${CASS_DRIVER_BUILD_CONCURRENCY}
+
+ARG AIRFLOW_VERSION
+ENV AIRFLOW_VERSION=${AIRFLOW_VERSION}
+
+ARG AIRFLOW_EXTRAS
+ENV AIRFLOW_EXTRAS=${AIRFLOW_EXTRAS}
+
+ARG AIRFLOW_INSTALL_SOURCES="."
+ENV AIRFLOW_INSTALL_SOURCES=${AIRFLOW_INSTALL_SOURCES}
+
+ARG AIRFLOW_INSTALL_VERSION=""
+ENV AIRFLOW_INSTALL_VERSION=${AIRFLOW_INSTALL_VERSION}
+
+ARG CONSTRAINT_REQUIREMENTS="requirements/requirements-python${PYTHON_MAJOR_MINOR_VERSION}.txt"
+ENV CONSTRAINT_REQUIREMENTS=${CONSTRAINT_REQUIREMENTS}
+
+ARG AIRFLOW_SOURCES_FROM="."
+ENV AIRFLOW_SOURCES_FROM=${AIRFLOW_SOURCES_FROM}
+
+WORKDIR /opt/airflow
+
+# hadolint ignore=DL3020
+ADD "${CONSTRAINT_REQUIREMENTS}" /requirements.txt
+
+ENV PATH=${PATH}:/root/.local/bin
+
+RUN pip install --user "${AIRFLOW_INSTALL_SOURCES}[${AIRFLOW_EXTRAS}]${AIRFLOW_INSTALL_VERSION}" \
+    --constraint /requirements.txt && \
+    find /root/.local/ -name '*.pyc' -print0 | xargs -0 rm -r && \
+    find /root/.local/ -type d -name '__pycache__' -print0 | xargs -0 rm -r
+
+
+ARG WWW_FOLDER
+ENV WWW_FOLDER=${WWW_FOLDER}
+
+ENV AIRFLOW_WWW=/root/.local/lib/python${PYTHON_MAJOR_MINOR_VERSION}/site-packages/airflow/${WWW_FOLDER}
+
+RUN yarn --cwd ${AIRFLOW_WWW} install --frozen-lockfile --no-cache \
+    && yarn --cwd ${AIRFLOW_WWW} run prod \
+    && rm -rf ${AIRFLOW_WWW}/node_modules
+
+ARG ENTRYPOINT_FILE="entrypoint.sh"
+ENV ENTRYPOINT_FILE="${ENTRYPOINT_FILE}"
```

Comment:
Still seems odd to me that this is an argument - I can't see when we'd chagne it?

Reply:
I checked how it is in Breeze - it is only needed in case we have different entrypoint.sh for different versions of Airflow. I am using the same Dockerfile to build 2.0 and 1.10 when installing from pypi or URL. for now there is no need indeed as I removed resetdb vs. dbreset -- so entrypoints will be the same but I want to keep the possibility that entrypoints will be different in 1.10 and 2.0 and that we would like to keep them separated. I really like (and yes) all the functionality to be able to build and run production-version of the released airflow in the master branch. That's really helpful to test 1.10 images.

It's the same thing as the requirements.txt - we get them from v1_10_test branch in order to build 1.10 image from master.  I think it's rather useful. I found it useful for my testing at least

Reply:
Do you see any security problems with it? Or do you have other concerns?

Reply:
No no problem with it, just wasn't quite sure what purpose it served.